src/main/java/com/yahoo/bullet/querying/QueryCategorizer.java

```diff
@@ -51,6 +53,19 @@ public QueryCategorizer categorize(BulletRecord record, Map<String, Querier> que
         return this;
     }
 
+    /**
+     * Categorize the given {@link Map} of query IDs to {@link Querier} instances after consuming the given {@link List}
+     * records. Invokes {@link #categorize(BulletRecord, Map)} for each record.
+     *
+     * @param records The {@link List} of {@link BulletRecord} to consume.
+     * @param queries The queries to categorize.
+     * @return This object for chaining.
+     */
+    public QueryCategorizer categorize(List<BulletRecord> records, Map<String, Querier> queries) {
```

Comment:
It seems not useful since every record has different queries to see.

Reply:
For this interface, all records see all the queries. The QuerierManager doesn't add a method for calling this categorize though after partition. Do we want that? I added this mostly for spark since bullet-spark won't use the QuerierManager but might want to use the QueryCategorizer.